src/protocol/libp2p/kademlia/mod.rs

```diff
@@ -636,11 +638,32 @@ impl Kademlia {
 
                 Ok(())
             }
-            QueryAction::GetRecordQueryDone { query_id, record } => {
-                self.store.put(record.record.clone());
+            QueryAction::GetRecordQueryDone { query_id, records } => {
+                // Considering this gives a view of all peers and their records, some peers may have
+                // outdated records. Store only the record which is backed by most
+                // peers.
+                let rec = records
+                    .iter()
+                    .map(|peer_record| &peer_record.record)
+                    .fold(HashMap::new(), |mut acc, rec| {
+                        *acc.entry(rec).or_insert(0) += 1;
+                        acc
+                    })
+                    .into_iter()
+                    .max_by_key(|(_, v)| *v)
+                    .map(|(k, _)| k);
```

Comment:
Would be nice to have tests for correctly updating the local store and also discarding the expired records. At least as a follow-up planned if it's too much work to implement such tests.